Source/Request.swift

```diff
@@ -1206,15 +1209,16 @@ public final class DataStreamRequest: Request {
     }
 
     func didReceive(data: Data) {
-        $streamMutableState.read { state in
+        $streamMutableState.write { state in
             if let stream = state.outputStream {
                 underlyingQueue.async {
                     var bytes = Array(data)
                     stream.write(&bytes, maxLength: bytes.count)
                 }
             }
-
-            underlyingQueue.async { state.streams.forEach { stream in stream.queue.async { stream.stream(data) } } }
```

Comment:
Looks like this is the underlying race condition that we're dispatching the streams on the `underlyingQueue` but the response serializer can complete while this is still running. Is the the crux of it?

Reply:
If we didn't dispatch here, would we eliminate the race? Probably not at all the right solution, just trying to better understand the race itself as to why you landed on this solution.

Reply:
No, eliminating this dispatch doesn't fix the race. Ultimately, the race lives in the `serializationQueue`, hence the added test. In that test case, the extra work of dispatching onto a separate, parallel queue results in about a 10% failure rate (with some weird behavior), as the completion event avoids that work altogether and can finish much more quickly.